planner/core/testdata/plan_suite_out.json

```diff
@@ -1432,6 +1432,182 @@
       }
     ]
   },
+  {
+    "Name": "TestPushdownDistinctEnable",
+    "Cases": [
+      {
+        "SQL": "select /*+ HASH_AGG() */ a, count(distinct a) from t;",
```

Comment:
Why we need this test?

Reply:
the `firstrow` for column `a` can not be removed.
See https://github.com/pingcap/tidb/pull/15500/files#diff-2f445d8cb9c2f7d8ec4fb0586ae1f510R1223-R1226

Reply:
Comments in code are added.